docs/docs/introducing-jsx.md

```diff
@@ -0,0 +1,202 @@
+
+```js{3,5}
+function getGreeting(user) {
+  if (user) {
+    return <h1>Hello, {formatName(user.name)}!</h1>;
+  } else {
+    return <h1>Hello, Stranger.</h1>;
+  }
+}
+```
+
+### Specifying Attributes with JSX
+
+You may use quotes to specify string literals as attributes:
+
+```js
+let element = <div tabIndex="0"></div>;
+```
+
+You may also use the curly braces to embed a JavaScript expression in an attribute:
+
+```js
+let element = <img src={user.avatarUrl}></img>;
+```
+
+### Specifying Children with JSX
+
+If a tag is empty, you may close it immediately with `/>`, like in XML:
+
+```js
+let element = <img src={user.avatarUrl} />;
+```
+
+JSX tags may contain children:
+
+```js
+let element = (
+  <div>
+    <h1>Hello!</h1>
+    <h2>Good to see you here.</h2>
+  </div>
+);
+```
+
+>**Caveat:**
+>
+>Since JSX is closer to JavaScript than HTML, it uses property names as defined in the DOM specification rather than HTML attribute names.
+>
+>For example, `class` becomes [`className`](https://developer.mozilla.org/en-US/docs/Web/API/Element/className) in JSX, and `tabindex` becomes [`tabIndex`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/tabIndex). The convention is to use `camelCase` attribute names, just like properties in JavaScript.
+
+### JSX Represents Objects
+
+Babel compiles JSX down to `React.createElement()` calls.
+
+These two examples are identical:
+
+```js
+let element = (
+  <h1 className="greeting">
+    Hello, world!
+  </h1>
+);
+```
+
+```js
+let element = React.createElement(
+  'h1',
+  {className: 'greeting'},
+  'Hello, world!'
+);
+```
+
+We will get to what "React elements" are in the next section. For now, you can think of them as descriptions of what you want to see on the screen.
+
+`React.createElement()` performs a few checks to help you write bug-free code but essentially it creates an object like this:
+
+```js
+// Note: this structure is simplified
+let element = {
+  type: 'h1',
+  props: {
+    className: 'greeting',
+    children: 'Hello, world'
+  }
+};
+```
+
+React later reads these objects and uses them to construct the DOM and keep it up to date.
+
+>**Caveat:**
+>
+>Every tag becomes a function call so you can't leave two tags without a parent.
+>
+>```js
+// JSX compilation error:
+// "Adjacent JSX elements must be wrapped in an enclosing tag"
+let element = (
+  <h1>Hello!</h1>
+  <h2>Good to see you here.</h2>
+);
+```
+>
+>We can fix it by adding a root tag to the JSX expression:
+>
+>```js{2,5}
+let element = (
+  <div>
+    <h1>Hello!</h1>
+    <h2>Good to see you here.</h2>
+  </div>
+);
+```
+
+### JSX Prevents Injection Attacks
+
+It is important to stress that JSX produces objects rather than string literals.
+
+This is why it is safe to embed user input in it:
+
+```js
+let title = response.potentiallyMaliciousInput;
+// This is safe:
+let element = <h1>{title}</h1>;
+```
+
+React will escape any values embedded in JSX before rendering them.
```

Comment:
Added a note to mention it later, I don't think it matters in introduction though.